src/argilla_llama_index/llama_index_handler.py

```diff
@@ -105,72 +111,71 @@ def __init__(  # noqa: C901
                 "Both `api_url` and `api_key` must be set. The current values are: "
                 f"`api_url`={api_url} and `api_key`={api_key}."
             )
+        self.client = Argilla(api_key=api_key, api_url=api_url)
 
-        client = rg.Argilla(api_key=api_key, api_url=api_url)
+        self.trace_buffer: List[SimpleSpan] = []
+        self.fields_info: Dict[str, Any] = {}
 
-        self.dataset_name = dataset_name
-        self.workspace_name = workspace_name
-        self.settings = rg.Settings(
+        self._initialize_dataset()
+
+    def _initialize_dataset(self):
+        """Create the dataset in Argilla if it does not exist, or update it if it does."""
+
+        self.settings = Settings(
             fields=[
-                rg.TextField(
-                    name="chat", title="Chat", use_markdown=True, required=True
-                ),
-                rg.TextField(
-                    name="time-details", title="Time Details", use_markdown=True
-                ),
+                TextField(name="chat", title="Chat", use_markdown=True, required=True),
+                # ChatField(name="chat", title="Chat", use_markdown=True, required=True), # TODO: Use ChatField when available.
             ]
-            + self._add_context_fields(number_of_retrievals),
+            + self._add_context_fields(self.number_of_retrievals)
```

Comment:
I would consider moving the document relevance score outside of the TextField/ChatField and create a separate Field. Also, I would perhaps create one or multiple metadata field to be able to filter using the retrieval scores.

Reply:
With the current implementation, you can filter by score according to the position of the document, i.e., you can filter by the score of all the second-retrieved documents. I'll add the scores as a different field.